src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -937,35 +948,33 @@ static int dhcp_server_relay_message(sd_dhcp_server *server, DHCPMessage *messag
         return -EBADMSG;
 }
 
-static int prepare_new_lease(
-                DHCPLease **ret_lease,
-                be32_t address,
-                const DHCPClientId *client_id,
-                uint8_t htype,
-                uint8_t hlen,
-                const uint8_t *chaddr,
-                be32_t gateway,
-                usec_t expiration) {
+static int prepare_new_lease(DHCPLease **ret_lease, be32_t address, DHCPRequest *req, usec_t expiration) {
```

Comment:
Mostly not your fault, but when you touch the code, please add assertions for arguments after the variable declaration below.
```c
assert(ret_lease);
assert(address != 0);
assert(req);
assert(expiration != 0);
```

Reply:
Done.